bin/process_bulk_metadata.py

```diff
@@ -0,0 +1,359 @@
+        #   ],
+        #   "abstract": "..."
+        # }
+        # ```
+
+        # why are these in there
+        issue_body = issue_body.replace("\r", "")
+
+        try:
+            match = re.search(r"```json\n(.*?)\n```", issue_body, re.DOTALL)
+
+            with open("test.json", "w") as f:
+                f.write(issue_body)
+
+            if match:
+                # return the first match
+                return json.loads(match[1])
+        except Exception as e:
+            print(f"Error parsing metadata changes: {e}", file=sys.stderr)
+
+        return None
+
+    def _apply_changes_to_xml(self, xml_path, anthology_id, changes):
+        """Apply the specified changes to XML file."""
+        try:
+            print(f"Applying changes to XML file {xml_path}", file=sys.stderr)
+            tree = ET.parse(xml_path)
+
+            collection_id, volume_id, paper_id = deconstruct_anthology_id(anthology_id)
+
+            paper_node = tree.getroot().find(
+                f"./volume[@id='{volume_id}']/paper[@id='{paper_id}']"
+            )
+            if paper_node is None:
+                print(f"-> Paper not found in XML file {xml_path}", file=sys.stderr)
+                return None
+
+            # Apply changes to XML
+            for key in ["title", "abstract"]:
+                if key in changes:
+                    node = paper_node.find(key)
+                    if node is None:
+                        node = make_simple_element(key, parent=paper_node)
+                    # set the node to the structure of the new string
+                    new_node = ET.fromstring(f"<{key}>{changes[key]}</{key}>")
+                    # replace the current node with the new node in the tree
+                    paper_node.replace(node, new_node)
+            if "authors" in changes:
+                """
+                Every author has an id, but for a small subset, these ids are explicit, since they're used for disambiguation. To distinguish these, we need to find the subset of the authors in the current XML that have explicit ID attributes. We then use this below to set the ID.
+                """
+                real_ids = set()
+                for author in changes["authors"]:
+                    id_ = author.get("id", None)
+                    if id_:
+                        existing_author = paper_node.find(f"author[@id='{id_}']")
+                        if existing_author is not None:
+                            real_ids.add(id_)
+
+                # remove existing author nodes
+                for author_node in paper_node.findall("author"):
+                    paper_node.remove(author_node)
+
+                prev_sibling = paper_node.find("title")
+
+                for author in changes["authors"]:
+                    attrib = {}
+                    if "id" in real_ids:
+                        # if the ID was explicitly represented, preserve it
+                        attrib["id"] = author["id"]
```

Comment:
I agree this is a possible situation. But my goal with this PR was to get something working and then refine it. Less than 0.5% of the `<author>` lines in the Anthology XML (1,908 / 354,582) have an id attribute. I'm not sure this is worth worrying about at this point; partly I was planning to revisit this next year when we figure out how to do explicit representation of people, rather than inferring people from names.

Reply:
Sure, I'm not saying this is a blocker. I would expect an explicit representation of people to mean that essentially _everyone_ has an ID, but of course we can wait until we have figured that out.

Reply:
I haven't thought it through completely, but that's loosely what I have in mind. And then, separately, there will be an in-repo person database indexed by those IDs. Ingestion would then have to create new person entries, and we'd have to occasionally clean them out if any were orphaned.